### PR TITLE
Extended lint applied

### PR DESCRIPTION
**Pull request by doktordirk**

### CODE REVIEW

File: src/repository.js [hunk 1]
```diff
@@ -219,13 +220,15 @@ export class Repository {
     let associations = entity.getMeta().fetch('associations');
 
     for (let property in associations) {
-      let assocMeta = associations[property];
+      if ({}.hasOwnProperty.call(associations, property)) {
```
Comment (doktordirk) [inline src/repository.js]: that's http://eslint.org/docs/rules/guard-for-in (together with http://eslint.org/docs/rules/no-prototype-builtins )

File: src/entity.js [hunk 2]
```diff
@@ -175,7 +176,7 @@ export class Entity {
 
     return this.getTransport()
       .update(this.getResource(), this.getId(), requestBody)
-      .then((updated) => response = updated)
+      .then(updated => { response = updated; })
```
Comment (doktordirk) [inline src/entity.js:179]: that's http://eslint.org/docs/rules/no-return-assign

**Comment by doktordirk** [comment]
@RWOverdijk

File: src/repository.js [hunk 3]
```diff
@@ -194,6 +194,7 @@ export class Repository {
       }
 
       let repository     = this.entityManager.getRepository(entityMetadata.fetch('associations', key).entity);
+
```
Comment (RWOverdijk) [inline src/repository.js:197]: Why the newline? Is that really a linting rule?

File: src/entity.js (hunk 2 above)
Comment (RWOverdijk) [reply src/entity.js:179]: Mhm. I kind of get it, but it's convenient here. So by rule, it should then be an assignment followed by a return?

Either way, I think we can lose the curly braces, and the semicolon?

File: src/decorator/idProperty.js [hunk 4]
```diff
@@ -1,5 +1,14 @@
 import {OrmMetadata} from '../orm-metadata';
 
+/**
+ * Set the id property for en entity
+ *
+ * @export
+ * @param {string} propertyName
+ * @returns {function}
+ *
+ * @deocator
```
Comment (RWOverdijk) [inline src/decorator/idProperty.js]: deocator?

File: src/component/association-select.js [hunk 5]
```diff
@@ -134,7 +135,8 @@ export class AssociationSelect {
       });
     }
 
-    return repository.findPath(findPath, criteria).catch(error => this.error = error);
+    return repository.findPath(findPath, criteria)
+      .catch(error => { this.error = error; });
```
Comment (RWOverdijk) [inline src/component/association-select.js]: Also return, in case it's going to be used. This is a BC break.

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/entity.js (hunk 2 above)
Comment (doktordirk) [reply src/entity.js:179]: that.s contradictory. either we use the rule and then we need curlies(+semicolons) or we allow returning assignments. unfortunatelly they don.t have an exception for fat arrows (yet)

File: src/repository.js (hunk 3 above)
Comment (doktordirk) [reply src/repository.js:197]: hmm. i might need to recheck.

File: src/decorator/idProperty.js (hunk 4 above)
Comment (doktordirk) [inline src/decorator/idProperty.js]: that s from auto jsdocs. admitably that s not very consistent. others include @export and @memberOf. might get useful for auto doc generation (which i wanted to recheck)

File: src/component/association-select.js (hunk 5 above)
Comment (doktordirk) [inline src/component/association-select.js]: so i guess we just disable the rule then

**Comment by doktordirk** [comment]
re: 

```
then(updated => { response = updated; })
      @RWOverdijk
RWOverdijk 3 hours ago
Mhm. I kind of get it, but it's convenient here. So by rule, it should then be an assignment followed by a return?
Either way, I think we can lose the curly braces, and the semicolon?
```

how?

File: src/repository.js (hunk 3 above)
Comment (doktordirk) [reply src/repository.js:197]: ah, ähm yes, that's ~ 'newline after variable declaration' or so (var, let , const)

File: src/decorator/idProperty.js (hunk 4 above)
Comment (doktordirk) [reply src/decorator/idProperty.js]: fixed

File: src/component/association-select.js (hunk 5 above)
Comment (doktordirk) [reply src/component/association-select.js]: fixed

File: src/entity.js (hunk 2 above)
Comment (RWOverdijk) [reply src/entity.js:179]: Allow return assignments :)

**Review by RWOverdijk — APPROVED** [review]

**Comment by RWOverdijk** [comment]
Allow return statements can be done later, wouldn't break current linting so no rush.

**Pull request merged**